airbyte-api/src/main/openapi/config.yaml

```diff
@@ -1756,13 +1756,21 @@ components:
           items:
             $ref: "#/components/schemas/SyncMode"
         sourceDefinedCursor:
-          description: If the source defines the cursor field, then it does any other cursor field inputs will be ignored. If it does not either the user_provided one is used or as a backup the default one is used.
+          description: If the source defines the cursor field, then any other cursor field inputs will be ignored. If it does not, either the user_provided one is used, or the default one is used as a backup.
           type: boolean
         defaultCursorField:
           description: Path to the field that will be used to determine if a record is new or modified since the last sync. If not provided by the source, the end user will have to specify the comparable themselves.
           type: array
           items:
             type: string
+        sourceDefinedPrimaryKey:
+          description: If the source defines the primary key, then any other primary key inputs will be ignored. If it does not, either the user_provided one is used, or the default one is used as a backup.
+          type: boolean
+        defaultPrimaryKey:
+          description: Path to the fields that will be used as a primary key. If not provided by the source, the end user will have to specify the primary key themselves.
```

Comment:
`defaultCursorField` is a bit mis-leading because it is one array representing one json path [a, b, c, field]: `{a: {b: {c: {field: value}}}}`

In your case you want more than one json path to define the primary key

Reply:
Really??

I thought it was an array of field names (string) from the top-level properties node of the stream, not that you could select a nested field as a cursor or primary key...
But in that case, what is the meaning of a cursor with json path `[a, b, c, field]` when the stream is:


```
{
    "a": {
        "b": [
            {
                "c": {
                    "field": "value1"
                }
            },
            {
                "c": {
                    "field": "value2"
                }
            }
        ]
    }
}```

Reply:
In this code, it doesn't seem like it is being used as a JSON path (to go into nested columns) but it is indeed an array of column names to form a composite column (which is not supported yet):

https://github.com/airbytehq/airbyte/blob/4c22b9b3444324f15ebfab3fde3a9004735654ed/airbyte-integrations/connectors/source-jdbc/src/main/java/io/airbyte/integrations/source/jdbc/IncrementalUtils.java#L36

Reply:
@michel-tricot 's interpretation is the the intention. the cursor field array is a json path. 

@ChristopheDuong agreed the error message you mentioned is misleading. it should say "nested" instead of "composite".

If you want to refer to multiple fields it would be `array<array<string>>` or equivalent.